pdfgen.c

```diff
@@ -2491,24 +2523,35 @@ static int pdf_add_jpeg_data(struct pdf_doc *pdf, struct pdf_object *page,
                              const unsigned char *jpeg_data, size_t len)
 {
     struct pdf_object *obj;
+    uint32_t img_width, img_height;
 
-    obj = pdf_add_raw_jpeg_data(pdf, jpeg_data, len);
+    obj = pdf_add_raw_jpeg_data(pdf, jpeg_data, len, &img_width, &img_height);
     if (!obj)
         return pdf->errval;
 
+    const int ret = get_img_display_dimensions(
+        img_width, img_height, &display_width, &display_height);
+    if (ret != 0) {
+        return ret;
```

Comment:
We need to set the pdf error stuff here, ie: `return pdf_set_err(pdf, ret, "invalid image dimensions")`

Reply:
I moved the error handling into `get_img_display_dimensions` by also passing in a pdf_doc*.
That way we can differentiate between the error cases and give a more meaningful error message to the user.